app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -2,30 +2,58 @@ package com.mapbox.vision.teaser
 
 import android.content.Intent
 import android.view.View
-import android.widget.ImageView
 import android.widget.LinearLayout
+import android.widget.Toast
 import com.mapbox.vision.VisionManager
+import com.mapbox.vision.VisionReplayManager
+import com.mapbox.vision.mobile.core.models.FrameStatistics
 import com.mapbox.vision.teaser.view.BaseTeaserActivity
 import com.mapbox.vision.teaser.view.show
 import com.mapbox.vision.safety.VisionSafetyManager
+import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.*
 import com.mapbox.vision.teaser.ar.ArMapActivity
+import com.mapbox.vision.teaser.replayer.ArReplayNavigationActivity
+import com.mapbox.vision.teaser.replayer.ReplayModeFragment
+import com.mapbox.vision.teaser.view.hide
 import com.mapbox.vision.view.VisionView
+import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : BaseTeaserActivity() {
+class MainActivity : BaseTeaserActivity(), ReplayModeFragment.OnClickModeItemListener {
+
+    enum class VisionManagerMode {
```

Comment:
I'd named it just `Mode`, since `VisionManagerMode`-`Replay` does not make sense - it uses `VisionReplayManager` instead

Reply:
Let's find a better name in next PRs where I've united the code. There is also AppMode for vision_view, would be nice to make both "Mode"s clear